tests/tensor/test_math.py

```diff
@@ -3731,3 +3732,36 @@ def test_nan_to_num(nan, posinf, neginf):
         out,
         np.nan_to_num(y, nan=nan, posinf=posinf, neginf=neginf),
     )
+
+
+@pytest.mark.parametrize(
+    "ndim, axis",
+    [
+        (2, None),
+        (2, 1),
+        (2, (0, 1)),
+        (3, None),
+        (3, (1, 2)),
+        (4, (1, 3, 0)),
+    ],
+)
+def test_median(ndim, axis):
+    # Generate random data with both odd and even lengths
+    shape = tuple(np.random.randint(2, 6) for _ in range(ndim))
```

Comment:
What about:
```python
shape_even = np.arange(1, ndim+1) * 2
shape_odd = shape_even - 1

data_even = np.random.rand(*shape_even)
data_odd = np.random.randl(*shape_odd)
```
I don't think you need the *100 either?

Reply:
the *100 was done so that the numbers are in range (0, 100) for convenience. Was easier to visualise the median than for numbers betweeen 0 to 1.  I'll rectify this.